compiler/atomic.go

```diff
@@ -73,6 +88,23 @@ func (b *builder) createAtomicOp(call *ssa.CallCommon) (llvm.Value, bool) {
 	case "StoreInt32", "StoreInt64", "StoreUint32", "StoreUint64", "StoreUintptr", "StorePointer":
 		ptr := b.getValue(call.Args[0])
 		val := b.getValue(call.Args[1])
+		if strings.HasPrefix(b.Triple, "avr") {
+			// SelectionDAGBuilder is currently missing the "are unaligned atomics allowed" check for stores.
+			vType := val.Type()
+			isPointer := vType.TypeKind() == llvm.PointerTypeKind
+			if isPointer {
+				// libcalls only supports integers, so cast to an integer.
```

Comment:
Technically they just don't care, they only see a bunch of bytes that they need to store. But that doesn't really matter.